apps/bfd-server/bfd-server-war/src/test/resources/openapi/headers.yaml

```diff
@@ -0,0 +1,162 @@
+        type: string
+      description: | 
+        The unique ID in the frontend system for the developer/vendor of the third-party application.
+    BlueButton-Developer-header:
+      in: header
+      name: BlueButton-Developer
+      required: false
+      schema:
+        type: string
+      description: |
+        The human-readable name in the frontend system for the developer/vendor of the third-party application.
+    BlueButton-ApplicationId-header:
+      in: header
+      name: BlueButton-ApplicationId
+      required: true
+      schema:
+        type: string
+      description: |
+        The unique ID in the frontend system for the application/client that will 
+        receive the data.
+    BlueButton-Application-header:
+      in: header
+      name: BlueButton-Application
+      required: false
+      schema:
+        type: string
+      description: | 
+        The human-readable name in the frontend system for the application/client that will 
+        receive the data.
+    BlueButton-UserId-header:
+      in: header
+      name: BlueButton-UserId
+      required: true
+      schema:
+        type: string
+      description: | 
+        The unique ID in the frontend system for the user (e.g. beneficiary) that data is being requested 
+        on behalf of.
+    BlueButton-User-header:
+      in: header
+      name: BlueButton-User
+      required: false
+      schema:
+        type: string
+      description: | 
+        The human-readable login ID (e.g. email address) in the frontend system for the user (e.g. beneficiary) 
+        that the data is being requested on behalf of.
+    BlueButton-BeneficiaryId-header:
+      in: header
+      name: BlueButton-BeneficiaryId
+      required: false
+      schema:
+        type: string
+      description: |
+        The unique beneficiary ID (e.g. Patient.id) from BFD for the user the beneficiary 
+        whose data is being requested.
+    x-forwarded-for-header:
+      in: header
+      name: x-forwarded-for
+      required: false
+      schema:
+        type: string
+      description: |
+        A standard header for identifying the originating IP address of a client connecting to 
+        a web server through an HTTP proxy or a load balancer.
+    BULK-CLIENTID-header:
+      in: header
+      name: BULK-CLIENTID
+      required: false
+      schema:
+        type: string
+      description: |
+        A unique identifier of the bulk client for whom this request is for (i.e. UUID, ACO ID, NPI, Part D Contract)
+    BULK-CLIENTNAME-header:
+      in: header
+      name: BULK-CLIENTNAME
+      required: false
+      schema:
+        type: string
+      description: A human readable name for the bulk client for whom this request is for to aid in tracing and debugging.
+    BULK-JOBID-header:
+      in: header
+      name: BULK-JOBID
+      required: false
+      schema:
+        type: string
+      description:  A unique identifier for the job (Job ID, UUID, etc.)
```

Comment:
While the bulk headers aren't required, I'd assume we'd want to mention they are HIGHLY desired (or maybe conditionally required) on bulk jobs so we can properly track them in our system

Reply:
They're either required or they aren't. Currently, they aren't. Even after this is published, I wouldn't expect anyone to read through all headers in the off-chance that the authors included a note urging that a header was _HIGHLY desired_. When this is sufficiently documented and socialized, we will begin planning for required inclusion of these headers, and update this documentation to be consistent with the implementation.